src/lease-process/lease-process-pool.test.ts

```diff
@@ -312,6 +335,7 @@ describe("LeaseProcessPool", () => {
       expect(pool["destroy"]).toHaveBeenCalledWith(lease3);
     });
     it("prevents new leases from being acquired during the drain", async () => {
+      jest.useFakeTimers();
```

Comment:
Isn't this a side-effect of enabling fake timers in that test above? IMO we should call `jest.useRealTImers` in `beforeEach` to "reset it"